ee/tabby-webserver/src/schema/github_repository_provider.rs

```diff
@@ -29,10 +29,38 @@ impl NodeType for GithubRepositoryProvider {
     }
 }
 
+#[derive(GraphQLObject, Debug)]
+#[graphql(context = Context)]
+pub struct GithubProvidedRepository {
+    pub id: ID,
+    pub vendor_id: String,
+    pub github_repository_provider_id: ID,
+    pub name: String,
+    pub git_url: String,
+    pub active: bool,
+}
+
+impl NodeType for GithubProvidedRepository {
+    type Cursor = String;
+
+    fn cursor(&self) -> Self::Cursor {
+        self.id.to_string()
+    }
+
+    fn connection_type_name() -> &'static str {
+        "GithubProvidedRepositoryConnection"
+    }
+
+    fn edge_type_name() -> &'static str {
+        "GithubProvidedRepositoryEdge"
+    }
+}
+
 #[async_trait]
 pub trait GithubRepositoryProviderService: Send + Sync {
     async fn get_github_repository_provider(&self, id: ID) -> Result<GithubRepositoryProvider>;
     async fn read_github_repository_provider_secret(&self, id: ID) -> Result<String>;
+    async fn read_github_repository_provider_access_token(&self, id: ID) -> Result<String>;
```

Comment:
just reuse `get_github_repository_provider`? You can still have `access_token` field in side of `GithubRepositoryProvider`, but need erase it before returning to client side.